src/header/internals/cell.rs

```diff
@@ -87,6 +87,20 @@ impl<V: ?Sized + Any + 'static> PtrMapCell<V> {
     }
 
     #[inline]
+    pub fn into_value(self, key: TypeId) -> Option<Box<V>> {
+        let mut map = unsafe { self.0.into_inner() };
```

Comment:
I was trying to keep this analogous to `get_mut` and `get`, but `UnsafeCell::into_inner` has [some extra caveats](https://doc.rust-lang.org/std/cell/struct.UnsafeCell.html#method.into_inner).